core/src/main/java/com/linecorp/armeria/common/Flags.java

```diff
@@ -1400,4 +1401,19 @@ private static <T> T getValue(Function<FlagsProvider, @Nullable T> method,
     }
 
     private Flags() {}
+
+    // These static variables are defined at the end of this file deliberately
+    // to ensure that all static variables beforehand are initialized.
+    private static final boolean INITIALIZED;
+    static {
+        INITIALIZED = true;
+    }
```

Comment:
https://stackoverflow.com/questions/12448465/in-what-order-are-static-blocks-and-static-variables-in-a-class-executed

This answer at SO helped me understand how class initialization works in more detail.

My understanding is since `boolean` is a `compile-time constant expression`, setting a value immediately like `private static final boolean INITIALIZED = true;` doesn't necessarily mean all static expressions in the file are evaluated beforehand.

I wanted to prevent any optimization the jvm might do, so I decided to define a static initializer separately.

![Screen Shot 2022-06-08 at 1 56 04 PM](https://user-images.githubusercontent.com/8510579/172534912-49ff916a-9a6a-48c4-8975-255f07731455.png)
![Screen Shot 2022-06-08 at 1 56 12 PM](https://user-images.githubusercontent.com/8510579/172534903-110829f0-68eb-4661-a4e9-129fad38e5a9.png)

Another thing I think I could've done is create a dummy `Object` and check for nullability. I thought the current way is more explicit though.

Ref: https://docs.oracle.com/javase/specs/jls/se7/html/jls-12.html#jls-12.4.2

Reply:
Thanks for the explanation. 😉
I slightly prefer to just have a dummy object with a good name but the current code also looks nice.